tests/scripts/thread-cert/Cert_5_3_05_RoutingLinkQuality.py

```diff
@@ -45,90 +47,93 @@ def setUp(self):
+        # 2 & 3
+        leader_rloc = self.nodes[LEADER].get_ip6_address(config.ADDRESS_TYPE.RLOC)
 
-        addrs = self.nodes[ROUTER3].get_addrs()
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertTrue(self.nodes[LEADER].ping(addr))
-
-        self.nodes[LEADER].add_whitelist(self.nodes[ROUTER1].get_addr64(), rssi=-95)
-        self.nodes[ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64(), rssi=-95)
+        # Verify the ICMPv6 Echo Request take the least cost path.
+        self.assertTrue(self.nodes[ROUTER3].ping(leader_rloc))
+        path = [ROUTER3, DUT_ROUTER1, LEADER]
+        command.check_icmp_path(self.sniffer, path, self.nodes)
 
+        # 4 & 5
+        self.nodes[LEADER].add_whitelist(self.nodes[DUT_ROUTER1].get_addr64(), config.RSSI['LINK_QULITY_1'])
+        self.nodes[DUT_ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64(), config.RSSI['LINK_QULITY_1'])
         time.sleep(70)
 
-        addrs = self.nodes[ROUTER3].get_addrs()
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertTrue(self.nodes[LEADER].ping(addr))
-
-        self.nodes[LEADER].add_whitelist(self.nodes[ROUTER1].get_addr64(), rssi=-85)
-        self.nodes[ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64(), rssi=-85)
+        # Verify the ICMPv6 Echo Request take the longer path because it cost less.
+        self.assertTrue(self.nodes[ROUTER3].ping(leader_rloc))
+        path = [ROUTER3, DUT_ROUTER1, ROUTER2, LEADER]
+        command.check_icmp_path(self.sniffer, path, self.nodes)
 
+        # 6 & 7
+        self.nodes[LEADER].add_whitelist(self.nodes[DUT_ROUTER1].get_addr64(), config.RSSI['LINK_QULITY_2'])
+        self.nodes[DUT_ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64(), config.RSSI['LINK_QULITY_2'])
         time.sleep(70)
 
-        addrs = self.nodes[ROUTER3].get_addrs()
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertTrue(self.nodes[LEADER].ping(addr))
-
-        self.nodes[LEADER].add_whitelist(self.nodes[ROUTER1].get_addr64(), rssi=-100)
-        self.nodes[ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64(), rssi=-100)
+        # Verify sending to a direct neighbor when have two paths with the same cost.
+        self.assertTrue(self.nodes[ROUTER3].ping(leader_rloc))
+        path = [ROUTER3, DUT_ROUTER1, LEADER]
+        command.check_icmp_path(self.sniffer, path, self.nodes)
 
+        # 8 & 9
+        self.nodes[LEADER].add_whitelist(self.nodes[DUT_ROUTER1].get_addr64(), config.RSSI['LINK_QULITY_0'])
+        self.nodes[DUT_ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64(), config.RSSI['LINK_QULITY_0'])
         time.sleep(70)
 
-        addrs = self.nodes[ROUTER3].get_addrs()
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertTrue(self.nodes[LEADER].ping(addr))
-
+        # Verify the ICMPv6 Echo Request take the least cost path.
```

Comment:
nit: take->took or takes ?
the least cost path -> the longer path ? (as said in Step 9).